File: tests/docs/test_diagram_attr.cpp
#include "svg_diagram.h"

#include <cmath>
#include <numbers>
#include <format>
#include <vector>
#include <gtest/gtest.h>
using namespace svg_diagram;

static const auto OUTPUT_DIR = std::string("../docs/source/_static/diagram_attr/");

TEST(TestDocsDiagramAttributes, BackgroundColor) {
    SVGDiagram diagram;
    diagram.setBackgroundColor("lightgray");
    const auto node = diagram.addNode("A");
    node->setLabel("A");
    diagram.render(OUTPUT_DIR + "background_color.svg");
}

File: python/tests/docs/test_diagram_attr.py
from sp_svg_diagram import SVGDiagram

from .utils import compare_svg


def test_background_color():
    diagram = SVGDiagram()
    diagram.set_background_color("lightgray")
    node = diagram.add_node("A")
    node.set_label("A")
    svg = diagram.render()
    compare_svg("diagram_attr", "background_color", svg)
